hoomd/hpmc/pytest/test_muvt.py

```diff
@@ -159,3 +159,144 @@ def test_insertion_removal(device, simulation_factory,
 
     # make a wild guess: there be B particles
     assert (muvt.N['B'] > 0)
+
+@pytest.mark.skipif(not hoomd.version.llvm_enabled, reason='LLVM not enabled')
+def test_jit_remove_insert(device, simulation_factory,
+                           one_particle_snapshot_factory):
+    """Test that MuVT considers a cpp potential when removing particles by adding a high energy particles that should always be removed by the updater while inserting particles to the top of the box."""
+    sim = simulation_factory(
+        one_particle_snapshot_factory(particle_types=['A'],
+                                 dimensions=3,
+                                 position=(0, 0, -5),
+                                 orientation=(1, 0, 0, 0),
+                                 L=20)
+                                    )
+
+    sphere_radius = 0.6 
+    mc = hoomd.hpmc.integrate.Sphere(default_d=0.0, default_a=0.0)
+    mc.shape['A'] = dict(diameter=2*sphere_radius)
+
+    # code returns infinity if the particle center goes past the center of the box
+
+    center_wall = '''
+        if (r_i.z < 0. || r_i.z > box.getL().z/2)
```

Comment:
Is the second part of this boolean expression necessary? Any z-position greater than L/2 should be outside the box.

Reply:
A trial move temporarily generates positions outside the box. For technical reaons, HPMC does not wrap the trial move until long after it is accepted. Tobias and I discussed this and I will consider wrapping positions before evaluating wall potentials when I refactor this code. For now, this check is needed.